### PR TITLE
Python. Homework #2. Task #3. Пяйве Олег.

### PR DESCRIPTION
**Pull request by pyaiveoleg**

### CODE REVIEW

File: homeworks/homework_2/task_3/smart_args.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import inspect
+from copy import deepcopy
+from typing import Callable
+
+
+def smart_args(function: Callable = None, positional_arguments_included: bool = False):
```
Comment (nbirillo) [inline homeworks/homework_2/task_3/smart_args.py]: Please, use `*` between `function` and `positional_arguments_included`

File: homeworks/homework_2/task_3/smart_args.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    To add deep copy of mutable argument, use Isolated() as default arg of your function.
+
+    You can add support of positional arguments using positional_arguments_included=True
+    """
+    if function is None:
+        return lambda f: smart_args(f, positional_arguments_included)
+
+    def inner(*args, **kwargs):
+        a = inspect.getfullargspec(function)
+        # print(a)
```
Comment (nbirillo) [inline homeworks/homework_2/task_3/smart_args.py]: Should you delete it?

File: homeworks/homework_2/task_3/smart_args.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    To add evaluation, use Evaluated(function) as default arg of your function
+    To add deep copy of mutable argument, use Isolated() as default arg of your function.
+
+    You can add support of positional arguments using positional_arguments_included=True
+    """
+    if function is None:
+        return lambda f: smart_args(f, positional_arguments_included)
+
+    def inner(*args, **kwargs):
+        a = inspect.getfullargspec(function)
```
Comment (nbirillo) [inline homeworks/homework_2/task_3/smart_args.py]: Please, don't use too short variable names

File: homeworks/homework_2/task_3/smart_args.py (hunk 1 above)
Comment (nbirillo) [reply homeworks/homework_2/task_3/smart_args.py]: Please, fix the following case:
```
@smart_args(positional_arguments_included=True)
def support_positionals(x=Isolated(), y=5, /):
    x[0] += 1
    return (x, y)

X = [0]
print(support_positionals(X, 5))
print(support_positionals(X))
```
I got an IndexError error

Comment (nbirillo) [reply homeworks/homework_2/task_3/smart_args.py]: And this one:
```
@smart_args(positional_arguments_included=True)
def support_positionals(x, y=Evaluated(lambda:123), /):
    x[0] += 1
    return (x, y)

X = [0]
print(support_positionals(X, 5))
print(support_positionals(X))
```
I got a TypeError

**Review by nbirillo — CHANGES_REQUESTED** [review]

**Review by SpirinEgor — APPROVED** [review]

**Pull request merged**